hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -1600,15 +1605,16 @@ void compactIfNecessary(BaseHoodieWriteClient<?,I,?,?> writeClient, Option<Strin
       LOG.info("Compaction with same {} time is already present in the timeline.", compactionInstantTime);
     } else if (writeClient.scheduleCompactionAtInstant(compactionInstantTime, Option.empty())) {
       LOG.info("Compaction is scheduled for timestamp {}", compactionInstantTime);
-      writeClient.compact(compactionInstantTime);
+      HoodieWriteMetadata<O> compactionWriteMetadata = writeClient.compact(compactionInstantTime);
+      writeClient.commitCompaction(compactionInstantTime, compactionWriteMetadata, Option.empty());
```

Comment:
shall we also keep the original `#commit` method for inline completion, it's kind of verbose to call 2 times here.

Reply:
yeah, for inline compaction, I can introduce a public method in WriteClient only. should be doable.